crates/sncast/tests/helpers/fixtures.rs

```diff
@@ -267,29 +267,44 @@ pub fn create_test_provider() -> JsonRpcClient<HttpTransport> {
 }
 
 #[must_use]
-pub fn duplicate_directory_with_salt(src_path: String, to_be_salted: &str, salt: &str) -> TempDir {
+pub fn duplicate_contract_directory_with_salt(
+    src_path: String,
+    to_be_salted: &str,
+    salt: &str,
+) -> TempDir {
+    duplicate_directory_and_salt_file(src_path, None, to_be_salted, "src/lib.cairo", salt)
+}
+
+#[must_use]
+pub fn duplicate_directory_and_salt_file(
+    src_path: String,
+    dst_path: Option<String>,
+    code_to_be_salted: &str,
+    file_to_be_salted: &str,
+    salt: &str,
+) -> TempDir {
     let src_dir = Utf8PathBuf::from(src_path);
-    let temp_dir = TempDir::new().expect("Unable to create a temporary directory");
+    let temp_dir = match dst_path {
+        Some(dst_path) => TempDir::new_in(dst_path),
```

Comment:
https://github.com/foundry-rs/starknet-foundry/pull/1358#discussion_r1472517520

Reply:
sure, so I guess this becomes a [nitpick] then, but I guess we can probably canonicalize these paths then, and make s path swap in file a part of copy function? should't be complicated, but not sure it's worth it though, that's why a nitpick

Reply:
I think we may consider refactoring this into two functions: `duplicate_contract_directory` and `duplicate_script_directory`. The last one will accept canonical paths to script dependencies (except for sncast_std which will be determined automatically)

So the flow would be:
1. duplicate all contract directories script depends on
2. duplicate script directory

Reply:
yes, I think this may be a good idea. let's do this.

Reply:
done